webapps/api/backend/decorators.py

```diff
@@ -289,11 +302,25 @@ def new_function(*args, **kwargs):
             guid_list = isinstance(data_list, list) and len(data_list) > 0 and isinstance(data_list[0], basestring)
             timings['fetch'] = [time.time() - start, 'Fetching data']
 
+            # Filtering data
```

Comment:
Why not apply the query right away in the `Fetch data` step?

Reply:
To make it pretty
The sorting could also be applied in that step really